hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/compact/FlinkScheduleCompactionActionExecutor.java

```diff
@@ -72,6 +75,17 @@ protected HoodieCompactionPlan scheduleCompaction() {
     boolean compactable = needCompact(config.getInlineCompactTriggerStrategy());
     if (compactable) {
       LOG.info("Generating compaction plan for merge on read table " + config.getBasePath());
+      // roll back the inflight compaction first
+      HoodieTimeline pendingCompactionTimeline = table.getActiveTimeline().filterPendingCompactionTimeline();
```

Comment:
would you please clarify why we need rollback inflight compaction here? Scheduling different compactions should be ok, is any other constraint in flink codebase?

Reply:
In flink compaction, if a merge fails during execution, our program will simply ignore it and this compaction won't end.
So we need to rollback the pending compaction when the scheduling the new compaction.

Reply:
@yuzhaojing So would we know the reason why compaction fails? and if we just rollback the inflight compaction and schedule a new one, the new one will possibly fail again?

Reply:
In my case is a hdfs cluster jitter，this is a occasional problems that we retry compaction.

Reply:
@leesf We can not assume each compaction in flink would succeed, the implementation for spark and flink are different.

Reply:
connected offline with @yuzhaojing . Our program will simply ignore it and does not fail will cause the filegroups in compaction plan never get compacted again if not delete inflight instant, we need delete the compaction instant in rollback to make the filegroups to be included in the next compaction plan.  the new change LGTM and will land after CI pass.

Reply:
Yes, so we should delete instant instead rollback.